### DISCUSSION TITLE
Claimed entities don't dissapear after owner inactivity

### DISCUSSION THREAD

**Discussion by MithrandirCraft** [GriefPrevention/GriefPrevention#260] [Ideas]
### What behaviour is observed:
Protected entities are still protected after the claims dissapear and the owner is long gone.
They should also be unclaimed after a certain owner inactivity, just like terrains.
### What behaviour is expected:
Well, I did expect that, but it would be nice if entity claims could also be affected by owner inactivity
### Steps/models to reproduce:
Any protected entity will not unprotect itself, no matter how long the user is inactive.

### Spigot/Craftbukkit/Paper and GriefPrevention version:
Spigot: CraftBukkit version git-Spigot-4bd94dc-4a03053 (MC: 1.12.2)
GP: 16.7.1

### Stack trace/error
No errors. It's a fault.

**Comment by RoboMWM** [comment]
> Any protected entity will not unprotect itself, no matter how long the user is inactive.

What protected entities? So the claim disappears but the entity is still protected?

**Comment by MithrandirCraft** [comment]
Any protectable mob entity, and yeah, since I don't have them configured for them to only be protected if they are inside an area, therefore, they remain protected forever.

**Comment by RoboMWM** [comment]
Still not sure what you mean. You mean the tameable mobs (horses, cats, and wolves in non-pvp worlds)?

In this case I don't think protection expires for these entities since claim expiration is just that, claim expiration. Though I do vaguely remember something regarding "untaming" though I don't recall if that was in here or in PopulationDensity. Unless you can find out that this was an intended feature I'm marking this as a feature request for now.

**Comment by MithrandirCraft** [comment]
ProtectCreatures: true
ProtectHorses: true
ProtectPetsOutsideLandClaims: true (I don't want them to be unprotected just because they are let 
loose)

Therefore, pets are eternally protected on my server due to this combination. This is a real issue, and nothing else, since it affects world performance, environment appearance, and resource abundance, negatively.
The fix I suggest (Allthough maybe not the most optimal) is unclaiming entities when owner is inactive, decidable by config. Just like normal terrain claims.

**Comment by RoboMWM** [comment]
Well it's not the same because these entities could be anywhere and are not stored in GP. What we'd have to do is store deleted player UUIDs and then check the owner of each entity that loads in each chunk load. That, or check on interaction. Either way, we're needing to store and check for probably forever. Unless you can think of a better way?

**Comment by RoboMWM** [comment]
I think knowing the claim mode is also important context for this. Moving to discussions, but will move back to issues if need be.

**Reply by Jikoo** [reply]
Is it possible to change categories to ideas? If not, oh well, but this was actually something I had implemented as a part of GPClaimExpiration. I hadn't closed the feature request because I hadn't tested it thoroughly yet.

My solution was to use interaction/damage to trigger a check. If the pet isn't being touched, it's probably not in the way, so there's no need to bog down the server checking it every time a chunk loads or whatever.

If a pet is inside a claim and its owner is eligible to lose it, the pet transfers to the claim's owner.
If a pet is outside of claims and its owner is eligible to lose it, it resets to wild state - its owner will be cleared, it will stand up if it is sitting, and it will be untamed unless it cannot be tamed via vanilla means.

**Reply by RoboMWM** [reply]
yes you should be able to transfer discussions to different categories

**Reply by RoboMWM** [reply]
![](https://i.imgur.com/59iWqzF.png)

**Reply by Jikoo** [reply]
That appears to be a higher permission level button, which doesn't really make sense - you'd think it'd be on par permission-wise with issue labels. Heck, I can even migrate issues to discussions, I just can't change category after I've done it. I'll slap that in the feedback for 'em.
